src/pages/index.js

```diff
@@ -93,34 +117,98 @@ function setFormInfo(nameSelector, detailsSelector) {
 }
 
 function createCard(data) {
-  const cardElement = new Card({ data, handleImageClick }, "#cards-template");
+  const cardElement = new Card(
+    { data, handleImageClick },
+    "#cards-template",
+    deleteCardConfirm,
+    handleCardLike,
+    handleCardUnlike
+  );
   return cardElement.generateCard();
 }
 
+function deleteCardConfirm(id, cardElement) {
+  deleteCardConfirmModal.open();
+  deleteCardConfirmModal.setSubmitAction(() => {
+    handleDeleteCard(id, cardElement);
+  });
+}
+
 function handleImageClick(imgData) {
   previewModal.open(imgData);
 }
 
+function handleCardLike(id) {
+  cardInfo.cardLike(id);
+}
+
+function handleCardUnlike(id) {
+  cardInfo.cardUnlike(id);
+}
+
+function handleDeleteCard(id, cardElement) {
+  cardInfo.deleteCard(id).then((res) => {
+    cardElement.remove();
+    cardElement = null;
+    deleteCardConfirmModal.removeProcessingMessage("Yes");
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.


**Please, correct it everywhere**